tests/drop_packets/test_drop_counters.py

```diff
@@ -0,0 +1,295 @@
+    """
+    # Clear SONiC counters
+    duthost.command("sonic-clear counters")
+    duthost.command("sonic-clear rifcounters")
+    send_packets(pkt, duthost, ptfadapter, ports_info["ptf_tx_port_id"])
+    if discard_group == "L2":
+        verify_drop_counters(duthost, ports_info["dut_iface"], GET_L2_COUNTERS, L2_COL_KEY)
+        ensure_no_l3_drops(duthost)
+    elif discard_group == "L3":
+        if COMBINED_L2L3_DROP_COUNTER:
+            verify_drop_counters(duthost, ports_info["dut_iface"], GET_L2_COUNTERS, L2_COL_KEY)
+            ensure_no_l3_drops(duthost)
+        else:
+            if not tx_dut_ports:
+                pytest.fail("No L3 interface specified")
+
+            verify_drop_counters(duthost, tx_dut_ports[ports_info["dut_iface"]], GET_L3_COUNTERS, L3_COL_KEY)
+            ensure_no_l2_drops(duthost)
+    elif discard_group == "ACL":
+        if not tx_dut_ports:
+            pytest.fail("No L3 interface specified")
+
+        time.sleep(ACL_COUNTERS_UPDATE_INTERVAL)
+        acl_drops = duthost.acl_facts()["ansible_facts"]["ansible_acl_facts"]["DATAACL"]["rules"]["RULE_1"]["packets_count"]
+        if acl_drops != PKT_NUMBER:
+            fail_msg = "ACL drop counter was not incremented on iface {}. DUT ACL counter == {}; Sent pkts == {}".format(
+                tx_dut_ports[ports_info["dut_iface"]], acl_drops, PKT_NUMBER
+            )
+            pytest.fail(fail_msg)
+        if not COMBINED_ACL_DROP_COUNTER:
+            ensure_no_l3_drops(duthost)
+            ensure_no_l2_drops(duthost)
+    else:
+        pytest.fail("Incorrect 'discard_group' specified. Supported values: 'L2' or 'L3'")
+
+
+def check_if_skip():
+    if SKIP_COUNTERS_FOR_MLNX:
+       SKIP_COUNTERS_FOR_MLNX = False
+       pytest.skip("Currently not supported on Mellanox platform")
+
+
+@pytest.fixture(scope='module')
+def do_test():
+    def do_counters_test(discard_group, pkt, ptfadapter, duthost, ports_info, sniff_ports, tx_dut_ports=None, comparable_pkt=None):
+        """
+        Execute test - send packet, check that expected discard counters were incremented and packet was dropped
+        @param discard_group: Supported 'discard_group' values: 'L2', 'L3', 'ACL'
+        @param pkt: PTF composed packet, sent by test case
+        @param ptfadapter: fixture
+        @param duthost: fixture
+        @param dut_iface: DUT interface name expected to receive packets from PTF
+        @param sniff_ports: DUT ports to check that packets were not egressed from
+        """
+        check_if_skip()
+        base_verification(discard_group, pkt, ptfadapter, duthost, ports_info, tx_dut_ports)
+
+        # Verify packets were not egresed the DUT
+        exp_pkt = expected_packet_mask(pkt)
+        testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=sniff_ports)
+
+    return do_counters_test
+
+
+def test_absent_ip_header(do_test, ptfadapter, duthost, setup, tx_dut_ports, pkt_fields, ports_info):
```

Comment:
This is the situation – we have drop_packets.py file which all test functions appear there. 
But, if one of the tests (wjh/drop_couters) would like to have a separate test case (for example, currently we have acl tests only for drop_countes), it will be able to do so by implementing a test_* function only on  it’s own file. 
We agreed to do it this way so it will be easy to add shared test cases as well as separate ones.